bootstrap.sh is broken on Linux with GCC 6 (for example, on Debian 9)

**Describe the bug**
A clear and concise description of what the bug is.

```
-- Build files have been written to: /root/vcpkg/buildtrees/_vcpkg/build
[0/2] Re-checking globbed directories...
[4/91] Building CXX object CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o
FAILED: CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o
/usr/bin/g++-6 -DVCPKG_BASE_VERSION=2021-05-05 -DVCPKG_USE_STD_FILESYSTEM=0 -DVCPKG_VERSION=unknownhash -I/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include -O3 -DNDEBUG -std=c++1z -MD -MT CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o -MF CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o.d -o CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o -c /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In function ‘void load_config(vcpkg::Files::Filesystem&)’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:123:55: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto locked_metrics = Metrics::g_metrics.lock();
                                                       ^
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In lambda function:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:186:48: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto metrics = Metrics::g_metrics.lock();
                                                ^
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In function ‘int main(int, const char* const*)’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:207:55: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto locked_metrics = Metrics::g_metrics.lock();
                                                       ^
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:235:48: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto metrics = Metrics::g_metrics.lock();
                                                ^
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = vcpkg::Metrics::Metrics]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:51:29:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = vcpkg::Metrics::Metrics]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = std::__cxx11::basic_string<char>]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:137:36:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >::LockGuardPtr(vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >::LockGuardPtr(vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = std::__cxx11::basic_string<char>]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = vcpkg::Chrono::ElapsedTimer]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:167:30:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = vcpkg::Chrono::ElapsedTimer]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
[9/91] Building CXX object CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o
ninja: build stopped: subcommand failed.
```

**Environment**
- OS: [e.g. Windows/Linux etc...]
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install xxxx
2. See error
Repro code when

**Expected behavior**
A clear and concise description of what you expected to happen.

**Failure logs** 
-(please attached failure logs)

**Additional context**
Add any other context about the problem here.

### I am installing vcpkg on Ubuntu Debian 9 and doing the command

./vcpkg/bootstrap-vcpkg.sh

it gives me this error

note: so download it in debian 9 `git clone https://github.com/microsoft/vcpkg`

--- a/scripts/bootstrap.ps1
+++ b/scripts/bootstrap.ps1
@@ -45,7 +45,7 @@ while (!($vcpkgRootDir -eq "") -and !(Test-Path "$vcpkgRootDir\.vcpkg-root"))
 
 Write-Verbose "Examining $vcpkgRootDir for .vcpkg-root - Found"
 
-& "$scriptsDir/tls12-download.exe" github.com "/microsoft/vcpkg-tool/releases/download/2021-08-03/vcpkg.exe" "$vcpkgRootDir\vcpkg.exe"
+& "$scriptsDir/tls12-download.exe" github.com "/microsoft/vcpkg-tool/releases/download/2021-08-12/vcpkg.exe" "$vcpkgRootDir\vcpkg.exe"
 Write-Host ""
 
 if ($LASTEXITCODE -ne 0)

--- a/scripts/bootstrap.sh
+++ b/scripts/bootstrap.sh
@@ -278,8 +278,8 @@ else
 fi
 
 # Do the build
-vcpkgToolReleaseTag="2021-08-03"
-vcpkgToolReleaseSha="d4c25be83a7685bc268254909796b9ef8e1549454a45b1154ab1d64f9846c91ce48c8a1cf00a864afb9a6dd878d82ceaea5981d53e57968196a9b6e90934fdef"
+vcpkgToolReleaseTag="2021-08-12"
+vcpkgToolReleaseSha="a8b64125fb065129f7db48b299d8868a0394aa8a5c45e7c6076773d4bab9ca4b4d6cc4459df0df3399a7241fa0d168cdd2ab6169e4d8b7e6f708a52f003265b1"
 vcpkgToolReleaseTarball="$vcpkgToolReleaseTag.tar.gz"
 vcpkgToolUrl="https://github.com/microsoft/vcpkg-tool/archive/$vcpkgToolReleaseTarball"
 baseBuildDir="$vcpkgRootDir/buildtrees/_vcpkg"

